data_model/src/account.rs

```diff
@@ -212,6 +244,19 @@ pub struct Account {
 
 impl Identifiable for Account {
     type Id = Id;
+
+    fn id(&self) -> &Self::Id {
+        &self.id
+    }
+}
+
+impl HasMetadata for Account {
+    fn metadata(&self) -> &crate::metadata::Metadata {
+        &self.metadata
+    }
+}
```

Comment:
We could consider making a proc derive macro for the `Identifiable`/`HasMetadata` if we'll rely on the convention that id field is named `id` and metadata field is named `metadata`

Reply:
AFAIK this is not true only for `NewRole` at the moment

Reply:
I was considering that myself. This would also resolve the issue with `PartialEq` and `PartialOrd`.

Reply:
yeah, except that I initially thought those would be implemented with a declarative macro. Not sure, about a particular macro implementation, but `Identifiable` and `PartialEq`/`PartialOrd` impls are rather closely related.

Reply:
We have an issue for it. Resolving